api/extensions/ext_celery.py

```diff
@@ -50,21 +50,48 @@ def __call__(self, *args: object, **kwargs: object) -> object:
     celery_app.set_default()
     app.extensions["celery"] = celery_app
 
+    schedule = get_schedule(app.config)
+
     imports = [
         "schedule.clean_embedding_cache_task",
         "schedule.clean_unused_datasets_task",
     ]
-    day = app.config.get("CELERY_BEAT_SCHEDULER_TIME")
+
     beat_schedule = {
         "clean_embedding_cache_task": {
             "task": "schedule.clean_embedding_cache_task.clean_embedding_cache_task",
-            "schedule": timedelta(days=day),
+            "schedule": schedule,
         },
         "clean_unused_datasets_task": {
             "task": "schedule.clean_unused_datasets_task.clean_unused_datasets_task",
-            "schedule": timedelta(days=day),
+            "schedule": schedule,
         },
     }
     celery_app.conf.update(beat_schedule=beat_schedule, imports=imports)
 
     return celery_app
+
+def get_schedule(app_config):
+    """Determine the schedule type based on configuration."""
+
+    # Fetch configuration values
+    scheduler_type = app_config.get("CELERY_BEAT_SCHEDULER_TYPE")  # Options: 'time' or 'cron'
+    day = app_config.get("CELERY_BEAT_SCHEDULER_TIME")
+    cron_expression = app_config.get("CELERY_BEAT_SCHEDULER_CRON_EXPRESSION")
+
+    # Determine the schedule based on the 'scheduler_type' value
+    if scheduler_type == "cron" and cron_expression:
```

Comment:
How about use `match scheduler_type : case` for guarding condiction of scheduler_type. 

And check the cron_expression inside the `cron` branch, throwing error when cron_expression is missing. It's way better than just falling to unsatisfied branch of other type.